int32 id
string action
Entity target
bool is_done